doc/nrf/gs_modifying.rst

```diff
@@ -84,55 +93,116 @@ The following CMake commands can be managed by SES, if they target the ``app`` l
+The header file contains the same kind of information, but with defines usable by source code.
+
+For more detailed information, see Zephyr's :ref:`zephyr:dt-guide`.
+
+.. _configure_application_sw:
+
+Software-related configuration
+------------------------------
+
+.. ncs-include:: guides/build/index.rst
+   :docset: zephyr
+   :dedent: 3
+   :start-after: Kconfig
+   :end-before: Information from devicetree is available to Kconfig,
+
+Information from devicetree is available to Kconfig, through the functions defined in :file:`zephyr/scripts/kconfig/kconfigfunctions.py`.
+
+The single :file:`.config` file in the :file:`<build_dir>/zephyr/` directory describes the entire software configuration of the constructed binary.
+
+For detailed information about configuration options, see :ref:`zephyr:application-kconfig` in the Zephyr documentation.
+
+Memory layout configuration
+---------------------------
+
+The :ref:`partition_manager` is specific to the |NCS|.
+If enabled, it provides the memory layout configuration.
+The layout is impacted by various elements, such as Kconfig configuration options or the presence of child images.
+Partition Manager ensures that all required partitions are in the correct place and have the correct size.
+
+Dynamically generated configurations are supported by default, but you can also specify :ref:`ug_pm_static`.
+
+After CMake has run, a single :file:`partitions.yml` file with the complete memory layout will have been created in the :file:`build` directory.
+This process also creates a set of header files that provides defines, which can be used to refer to memory layout elements.
+
+Child images
+------------
+
+The |NCS| build system allows the application project to become a root for the sub-applications known in the |NCS| as :ref:`child images <ug_multi_image>`.
+Examples of child images are bootloader images, network core images, or security-related images.
+Each child image is a separate application.
 
 Changing the configuration temporarily
 ======================================
 
-When building your application, the different :file:`Kconfig` and :file:`*_defconfig` files and the :file:`prj.conf` file are merged together.
-The combined configuration is saved in a :file:`zephyr/.config` file in your build directory.
-This means that this file is available when building the application, but it is deleted when you clean the build directory.
+When building your application, the different :file:`Kconfig` and :file:`*_defconfig` files and the :file:`prj.conf` file are merged together and then processed by Kconfig.
+The resulting configuration is written to the :file:`zephyr/.config` file in your :file:`build` directory.
+This means that this file is available when building the application, but it is deleted when you clean the build directory with the ``pristine`` target (see Zephyr's :ref:`zephyr:application_rebuild` for more information).
```

Comment:
I find it unfortunate that the `pristine` target is highlighted exactly the same way as filenames are highlighted.

That is confusing when reading a section where both are used so often.
`pristine` is not a file, so could we have some another highlighting type for targets ?
![image](https://user-images.githubusercontent.com/9802583/145198266-e32eb592-d608-4c68-bbae-8cd59fb0e5de.png)

Reply:
Will follow this up with the doc framework team.